altsrc/map_input_source.go

```diff
@@ -71,24 +71,29 @@ func (fsm *MapInputSource) Int(name string) (int, error) {
 func (fsm *MapInputSource) Duration(name string) (time.Duration, error) {
 	otherGenericValue, exists := fsm.valueMap[name]
 	if exists {
-		otherValue, isType := otherGenericValue.(time.Duration)
-		if !isType {
-			return 0, incorrectTypeForFlagError(name, "duration", otherGenericValue)
-		}
-		return otherValue, nil
+		return castDuration(name, otherGenericValue)
 	}
 	nestedGenericValue, exists := nestedVal(name, fsm.valueMap)
 	if exists {
-		otherValue, isType := nestedGenericValue.(time.Duration)
-		if !isType {
-			return 0, incorrectTypeForFlagError(name, "duration", nestedGenericValue)
-		}
-		return otherValue, nil
+		return castDuration(name, nestedGenericValue)
 	}
 
 	return 0, nil
 }
 
+func castDuration(name string, value interface{}) (time.Duration, error) {
+	otherValue, isType := value.(time.Duration)
+	if !isType {
```

Comment:
Nitpick, but can we do an early return like

```go
if isType {
    return otherValue, nil
}

...
```

Just to remove some of the nesting.

Reply:
Agreed, done.